### PR TITLE
[BUGFIX beta] Fixes #12899

### LINKED ISSUE #12899: transitionTo returns promise when passed a url
In ember canary, [route.transitionTo](http://emberjs.com/api/classes/Ember.Route.html#method_transitionTo) returns a promise when passed a url. According to the docs, it should return a transition object.

From what I can tell, the issue seems to have been [introduced in this commit](https://github.com/emberjs/ember.js/commit/ef7108a8d1a97fc263918a0fd37db72889419b4c#diff-985dc4a078da4636eff18be21aa7f9e6L303).

I reproduced the issue in this [example app](https://github.com/asakusuma/transition-to-bug).

cc @minasmart @rwjblue

### PR DESCRIPTION
**Pull request by asakusuma**
Ensure transitionTo returns a transition object. Addresses https://github.com/emberjs/ember.js/issues/12899

### CODE REVIEW

**Comment by rwjblue** [comment]
Sorry for not getting this done quicker, but thank you for tackling it!

File: packages/ember-routing/lib/system/router.js [hunk 1]
```diff
@@ -303,9 +303,26 @@ var EmberRouter = EmberObject.extend(Evented, {
...
+
+    @method transitionTo
+    @param {String} name the name of the route or a URL
+    @param {...Object} models the model(s) or identifier(s) to be used while
+      transitioning to the route.
+    @param {Object} [options] optional hash with a queryParams property
+      containing a mapping of query parameters
+    @return {Transition} the transition object associated with this
+      attempted transition
+    @public
```
Comment (rwjblue) [inline packages/ember-routing/lib/system/router.js]: Lets land this fix without changing public API (docs here are great, but change `@public` to `@private`), then submit a separate documentation only PR that updates?

**Comment by rwjblue** [comment]
:+1:

**Comment by asakusuma** [comment]
No problem, @krisselden got me on the right approach

**Comment by rwjblue** [comment]
Yeah, he is awesome like that.  :smiley_cat: 

I need to determine if this bug is in 2.3.1, and backport to release if it is....

**Pull request merged**